source/common/network/address_impl.cc

```diff
@@ -398,16 +425,18 @@ PipeInstance::PipeInstance(const std::string& pipe_path, mode_t mode) : Instance
 bool PipeInstance::operator==(const Instance& rhs) const { return asString() == rhs.asString(); }
 
 Api::SysCallIntResult PipeInstance::bind(os_fd_t fd) const {
-  if (!abstract_namespace_) {
-    // Try to unlink an existing filesystem object at the requested path. Ignore
-    // errors -- it's fine if the path doesn't exist, and if it exists but can't
-    // be unlinked then `::bind()` will generate a reasonable errno.
-    unlink(address_.sun_path);
+  Api::OsSysCalls& os_sys_calls = Api::OsSysCallsSingleton::get();
+  if (abstract_namespace_) {
+    return os_sys_calls.bind(fd, reinterpret_cast<const sockaddr*>(&address_),
```

Comment:
Agreed, there is a much simpler way following this code pattern. I believe the change dates to long before the cleaner sockaddr accessors, since windows API uses a number of incomplete structs for these things, and the sizeof sockaddr_storage is generally preferred (except in this case.) Fixing in the next refresh.

Reply:
You were correct, the next commit simplifies this change.

Reply:
resolving, we re-simplified this code